java/test/problems/problem2/NumberOfMarksCost.java

```diff
@@ -31,25 +31,30 @@ public static void main(String args[]) {
         System.out.println("#       Measurement of search times: ");
         System.out.println("#        Marks from 0 to "+MAX_ARRAY+"    ");
         System.out.println("#----------------------------------------- ");
-        System.out.println("#  Size    Search time   Nº Marks > "+MARK+"   ");
+        System.out.println("#  Size    Search time                     ");
 
 
         for (ExamMarks ex : list){
             String clazz = ex.getClass().getName();
             t1 = t2 = time = 0;
-            System.out.println("#-----------------------------------------");
+            System.out.println("#--------------------------------------------------");
             System.out.println("#  \t\t" + clazz.replaceFirst("problems.problem2.", "") + " implementation.");
-            System.out.println("#-----------------------------------------");
+            System.out.println("#--------------------------------------------------");
             for (int t = SIZE_INI; t <= SIZE_FIN; t += SIZE_INCR) {
+                //The mark should be random and not predefined
```

Comment:
@josegarm Why mark changes in every time? 

The mark must be the same in each loop, isn't it? @carmocca 

Because if you have a random mark, in each loop it might be to length 40000 take less time than length 2000 because the mark is at the middle of array.